### ISSUE TITLE
Observable.webSocket should catch constructor errors

### ISSUE THREAD

**Issue by blakef** [ReactiveX/rxjs#1891]
**RxJS version:**
* 5.0.0-beta.11

**Code to reproduce:**
```javascript
const obs = Rx.Observable.webSocket('malformed_url');

obs.subscribe(
  ::console.log,
  err => console.error('It make sense that we _should_ receive an error here', err)
);
```
**Expected behavior:**

I'd expect to see and handle the error on the observable.

**Actual behavior:**

Throws a `DOMException: Failed to construct 'WebSocket': The URL 'malformed_url' is invalid` error.

**Additional information:**

I think it's worth discussing the design here.  For my particular use case, the url is being dynamically generated (I don't think this is unusual), so it makes sense to handle the error on the observable.

Thoughts?  I've drafted a test and fix.

**Comment by kwonoj** [comment]
For me this looks similar case to https://github.com/ReactiveX/rxjs/issues/1810, agree it'd better emit error instead of throwing out.

**Comment by kwonoj** [comment]
closed via #1895 .
